open-sphere-base/mantle/src/main/java/io/opensphere/mantle/plugin/selection/BufferRegionImpl.java

```diff
@@ -0,0 +1,320 @@
+package io.opensphere.mantle.plugin.selection;
+
+import java.awt.Frame;
+import java.awt.MouseInfo;
+import java.awt.Point;
+import java.util.Collections;
+
+import javax.swing.JOptionPane;
+import javax.swing.SwingUtilities;
+import javax.swing.event.PopupMenuEvent;
+
+import io.opensphere.core.Toolbox;
+import io.opensphere.core.control.ContextMenuSelectionAdapter;
+import io.opensphere.core.control.action.ActionContext;
+import io.opensphere.core.control.action.context.ContextIdentifiers;
+import io.opensphere.core.control.action.context.GeometryContextKey;
+import io.opensphere.core.geometry.Geometry;
+import io.opensphere.core.geometry.PolygonGeometry;
+import io.opensphere.core.units.UnitsProvider;
+import io.opensphere.core.units.length.Kilometers;
+import io.opensphere.core.units.length.Length;
+import io.opensphere.core.util.jts.core.JTSCoreGeometryUtilities;
+
+/**
+ * Creates a buffer region.
+ *
+ */
+public class BufferRegionImpl
```

Comment:
This isn't an implementation of an interface, so let's not call it Impl.  Maybe something like BufferRegionCreator.